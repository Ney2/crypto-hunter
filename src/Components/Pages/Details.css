* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.symbol-container {
  display: flex;
  justify-content: space-between;
  margin: 2%;
}

.symbol-info {
  text-align: right;
  margin-right: 20%;
  margin-top: -20%;
}

.arrow {
  left: 5%;
}

table {
  display: flex;
  justify-content: space-between;
  margin-top: 2%;
}

.title {
  text-align: left;
  padding-left: 5%;
}

.result {
  text-align: right;
  padding-right: 5%;
}

.details-header h1 {
  color: #ffd700;
  font-weight: bolder;
}

td {
  width: 100vw;
  height: 10vh;
  border: 1px solid #fff;
}

.details-header {
  text-align: center;
  margin-top: 20%;
  margin-bottom: 3%;
}

@media (min-width: 768px) {
  .symbol-container {
    display: flex;
    justify-content: space-between;
  }

  .symbol-container img {
    width: 30%;
    height: 30%;
  }

  .title {
    padding-left: 5%;
    text-align: left;
  }

  .result {
    text-align: right;
    padding-right: 5%;
  }

  .details-header h1 {
    color: #ffd700;
  }

  td {
    border: 1px solid #fff;
  }
}
