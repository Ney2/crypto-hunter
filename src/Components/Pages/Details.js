import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { useParams, useNavigate } from 'react-router-dom';
import * as BsIcon from 'react-icons/bs';
import { singleCoin } from '../../Redux/cryptoData';
import './Details.css';

const Details = () => {
  const { id } = useParams();
  const [coin, setCoin] = useState();

  const navigate = useNavigate();
  const navigateHistoricalChart = () => {
    navigate(`/details/histroicalchart/${id}`);
  };

  const fetchCoin = async () => {
    const { data } = await axios.get(singleCoin(id));
    setCoin(data);
  };

  useEffect(() => {
    fetchCoin();
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <section>
      <div className="symbol-container">
        <img className="pic" src={coin?.image.large} alt={coin?.name} />
        <BsIcon.BsFillArrowRightCircleFill className="arrow" id={id} onClick={navigateHistoricalChart} key={id} />
      </div>
      <div className="symbol-info">
        <p>
          Name:
          {' '}
          {coin?.name}
        </p>
        <p>
          Symbol:
          {' '}
          {coin?.symbol}
        </p>
        <p>
          Rank:
          {' '}
          {coin?.market_cap_rank}
        </p>
      </div>
      <div className="details-header">
        <h1>
          Details About
          { ' '}
          {coin?.name}
          { ' '}
          Crypto
        </h1>
      </div>
      <table>
        <tbody>
          <tr>
            <td className="title">Current Price</td>
            <td className="result">{coin?.market_data.current_price.usd}</td>
          </tr>
          <tr>
            <td className="title">Public Interest Score</td>
            <td className="result">{coin?.public_interest_score}</td>
          </tr>
          <tr>
            <td className="title">Liquidity Score</td>
            <td className="result">{coin?.liquidity_score}</td>
          </tr>
          <tr>
            <td className="title">Ath Percentage Change</td>
            <td className="result">{coin?.market_data.ath_change_percentage.usd}</td>
          </tr>
          <tr>
            <td className="title">Ath</td>
            <td className="result">{coin?.market_data.ath.usd}</td>
          </tr>
          <tr>
            <td className="title">High Market Data of 24h</td>
            <td className="result">{coin?.market_data.high_24h.usd}</td>
          </tr>
          <tr>
            <td className="title">Low Market Data of 24h</td>
            <td className="result">{coin?.market_data.low_24h.usd}</td>
          </tr>
          <tr>
            <td className="title">Market Cap 24h Change</td>
            <td className="result">{coin?.market_data.market_cap_change_24h}</td>
          </tr>
          <tr>
            <td className="title">Circulating Supply</td>
            <td className="result">{coin?.market_data.circulating_supply}</td>
          </tr>
          <tr>
            <td className="title">Market Cap 24h Percentage Change</td>
            <td className="result">{coin?.market_data.market_cap_change_percentage_24h}</td>
          </tr>
          <tr>
            <td className="title">Max Supply</td>
            <td className="result">{coin?.market_data.max_supply}</td>
          </tr>
          <tr>
            <td className="title">Price Change 24h</td>
            <td className="result">{coin?.market_data.price_change_24h}</td>
          </tr>
          <tr>
            <td className="title">Sentiment Votes Up Percentage</td>
            <td className="result">{coin?.sentiment_votes_up_percentage}</td>
          </tr>
          <tr>
            <td className="title">Sentiment Votes Down Percentage</td>
            <td className="result">{coin?.sentiment_votes_down_percentage}</td>
          </tr>
        </tbody>
      </table>
    </section>
  );
};

export default Details;
