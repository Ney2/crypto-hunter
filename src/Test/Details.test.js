import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Details from '../Components/Pages/Details';

test('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
