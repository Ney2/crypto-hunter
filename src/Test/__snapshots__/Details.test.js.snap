// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <div
    className="symbol-container"
  >
    <img
      className="pic"
    />
    <svg
      className="arrow"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
      />
    </svg>
  </div>
  <div
    className="symbol-info"
  >
    <p>
      Name:
       
    </p>
    <p>
      Symbol:
       
    </p>
    <p>
      Rank:
       
    </p>
  </div>
  <div
    className="details-header"
  >
    <h3>
      Details About
       
       
      Crypto
    </h3>
  </div>
  <table>
    <tbody>
      <tr>
        <td
          className="title"
        >
          Current Price
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Public Interest Score
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Liquidity Score
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Ath Percentage Change
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Ath
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          High Market Data of 24h
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Low Market Data of 24h
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Market Cap 24h Change
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Circulating Supply
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Market Cap 24h Percentage Change
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Max Supply
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Price Change 24h
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Sentiment Votes Up Percentage
        </td>
        <td
          className="result"
        />
      </tr>
      <tr>
        <td
          className="title"
        >
          Sentiment Votes Down Percentage
        </td>
        <td
          className="result"
        />
      </tr>
    </tbody>
  </table>
</section>
`;
